### PR TITLE
feat: horizontal overflow smooth scrolling

### PR DESCRIPTION
**Pull request by scomea**
# Description
This started as a fix to some browsers not scrolling focused elements into view when user tabbed to them, but that seems to be getting fixed in the browser (chromium at least).  This change should address any remnants of that issue but additionally it significantly modifies the scroll behavior to the component such that content should always smooth scroll into view even if users navigate through content quickly either by clicking the previous/next buttons or rapidly tabbing.

This change also adds a configurable "peek" property which moves the focused item away from the edge of the viewport if there is available space such that the next/previous item is partially visible.

## Motivation & context
https://github.com/microsoft/fast-dna/issues/2126

And a desire to improve how our components animate to create a premium experience.

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-base/src/horizontal-overflow/README.md [hunk 1]
```diff
@@ -4,5 +4,17 @@ The *horizontal overflow* component accepts items as children. A child can have
 ### Usage guidance
 The *horizontal overflow* component always initializes at the start of the overflow content &mdash; left side in LTR and right side in RTL. Several callbacks are supplied to provide the consumer with overflow and scroll data. The not fully supported [ResizeObserver](https://developers.google.com/web/updates/2016/10/resizeobserver) is used (supported in Chrome), so it is necessary to apply a polyfill if more thorough browser support is needed (Safari and Firefox). Full adaptation is expected soon.
 
+The "onOverflowChange" callback is provided to enable developers to react to changes in overflow.  The retured "OverflowChange" object describes overflow as follows
+ - When both "onOverflowChange" and "overflowEnd" are false, there is no overflow
+ - When both are true, there is overflow on either side
+ - 'overflowStart' is true when there are items to the left in LTR (right in RTL)
+ - 'overflowEnd' is true when there are items to the right in LTR (left in RTL)
+
+ The "onScrollChange" callback is expected to be depracated in a future release and should be avoided, use "onOverflowChange" instead.
```
[the remark this answers is not in the document]
Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/README.md]: That's one of the words that always gets me

File: packages/fast-components-react-base/src/horizontal-overflow/README.md [hunk 2]
```diff
@@ -4,5 +4,17 @@ The *horizontal overflow* component accepts items as children. A child can have
 ### Usage guidance
 The *horizontal overflow* component always initializes at the start of the overflow content &mdash; left side in LTR and right side in RTL. Several callbacks are supplied to provide the consumer with overflow and scroll data. The not fully supported [ResizeObserver](https://developers.google.com/web/updates/2016/10/resizeobserver) is used (supported in Chrome), so it is necessary to apply a polyfill if more thorough browser support is needed (Safari and Firefox). Full adaptation is expected soon.
 
+The "onOverflowChange" callback is provided to enable developers to react to changes in overflow.  The returned "OverflowChange" object describes overflow as follows:
+ - When both "onOverflowChange" and "overflowEnd" are `false`, there is no overflow
```
Comment (janechu) [inline packages/fast-components-react-base/src/horizontal-overflow/README.md]: did you mean onOverflowStart?

Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/README.md]: doh

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx [hunk 3]
```diff
@@ -41,14 +41,17 @@ class HorizontalOverflow extends Foundation<
 
     public static defaultProps: Partial<HorizontalOverflowProps> = {
         managedClasses: {},
+        nextItemPeek: 50,
     };
     private static DirectionAttributeName: string = "dir";
+    private static defaultScrollAnimDuration: number = 500;
```
[the remark this answers is not in the document]
Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: yep

File: packages/fast-components-react-base/src/horizontal-overflow/README.md [hunk 4]
```diff
@@ -4,5 +4,17 @@ The *horizontal overflow* component accepts items as children. A child can have
 ### Usage guidance
 The *horizontal overflow* component always initializes at the start of the overflow content &mdash; left side in LTR and right side in RTL. Several callbacks are supplied to provide the consumer with overflow and scroll data. The not fully supported [ResizeObserver](https://developers.google.com/web/updates/2016/10/resizeobserver) is used (supported in Chrome), so it is necessary to apply a polyfill if more thorough browser support is needed (Safari and Firefox). Full adaptation is expected soon.
 
+The "onOverflowChange" callback is provided to enable developers to react to changes in overflow.  The returned "OverflowChange" object describes overflow as follows:
+ - When both "onOverflowStart" and "overflowEnd" are `false`, there is no overflow
+ - When both are `true`, there is overflow on either side
+ - 'overflowStart' is `true` when there are items to the left in LTR (right in RTL)
+ - 'overflowEnd' is `true` when there are items to the right in LTR (left in RTL)
+
+ The "onScrollChange" callback is expected to be deprecated in a future release and should be avoided, use "onOverflowChange" instead.
```
[the remark this answers is not in the document]
Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/README.md]: Yes, deprecated now, to be removed later.  Updated.

File: packages/fast-components-react-base/src/horizontal-overflow/README.md [hunk 5]
```diff
@@ -4,5 +4,17 @@ The *horizontal overflow* component accepts items as children. A child can have
 ### Usage guidance
 The *horizontal overflow* component always initializes at the start of the overflow content &mdash; left side in LTR and right side in RTL. Several callbacks are supplied to provide the consumer with overflow and scroll data. The not fully supported [ResizeObserver](https://developers.google.com/web/updates/2016/10/resizeobserver) is used (supported in Chrome), so it is necessary to apply a polyfill if more thorough browser support is needed (Safari and Firefox). Full adaptation is expected soon.
 
+The "onOverflowChange" callback is provided to enable developers to react to changes in overflow.  The returned "OverflowChange" object describes overflow as follows:
+ - When both "onOverflowStart" and "overflowEnd" are `false`, there is no overflow
```
[the remark this answers is not in the document]
Comment (scomea) [reply packages/fast-components-react-base/src/horizontal-overflow/README.md]: yep

**Pull request merged**